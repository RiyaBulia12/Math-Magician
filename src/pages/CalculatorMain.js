import React from 'react';
import Calculator from '../components/Calculator';
import '../components/Calculator.css';

const CalculatorMain = () => (
  <div className="container">
    <h2>Let&apos;s do same maths!</h2>
    <Calculator />
  </div>
);
export default CalculatorMain;
