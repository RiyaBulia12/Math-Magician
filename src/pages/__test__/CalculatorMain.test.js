import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalculatorMain from '../CalculatorMain';

describe('Calculator Main Page', () => {
  afterEach(cleanup);

  it('renders calculator main page correctly', () => {
    const tree = renderer.create(<CalculatorMain />);
    expect(tree).toMatchSnapshot();
  });

  it('should display the calculator component on the page', () => {
    render(<CalculatorMain />);
    const calcContainer = document.getElementsByClassName('calc-container');
    expect(calcContainer).toBeTruthy();
  });
});
