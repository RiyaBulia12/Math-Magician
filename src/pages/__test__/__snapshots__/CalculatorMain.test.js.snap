// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Main Page renders calculator main page correctly 1`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator Main Page renders calculator main page correctly 2`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator Main Page should display content properly 1`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator Main Page should display the calculator component on the page 1`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator renders calculator main page correctly 1`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`renders calculator main page correctly 1`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <h2>
    Let's do same maths!
  </h2>
  <div
    className="calc-container"
  >
    <span
      className="answer"
    >
      0
    </span>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn last-col"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
